typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,541 @@
+                    resolve(okAsync(response));
+                }
+            })
+        });
+        const resultGetLambdaInfo: Result<GetLambdaInfoResponse, Error> = await promiseGetLambdaInfo;
+        if (!resultGetLambdaInfo.isOk()) {
+            return err(resultGetLambdaInfo.error);
+        }
+
+        const lambdaCtx: LambdaContext = new LambdaContext(this.client, lambdaId);
+        return ok(lambdaCtx);
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async registerStaticFiles(staticFileFilepaths: Map<StaticFileID, string>): Promise<Result<null, Error>> {
+        const strSet: Map<string, boolean> = new Map();
+        for (let [staticFileId, srcAbsFilepath] of staticFileFilepaths.entries()) {
+            
+            // Sanity-check that the source filepath exists
+            const promiseSrcAbsFilepath: Promise<fs.Stats> = fsPromises.stat(srcAbsFilepath);
+            promiseSrcAbsFilepath.catch(() => { 
+                return err(new Error("Source filepath " + srcAbsFilepath + " associated with static file " + staticFileId + " doesn't exist")); 
+            })
+            
+            strSet[String(staticFileId)] = true;
+        }
+
+        const args: RegisterStaticFilesArgs = newRegisterStaticFilesArgs(strSet);
+        
+        const promiseRegisterStaticFiles: Promise<ResultAsync<RegisterStaticFilesResponse, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.registerStaticFiles(args, (error: grpc.ServiceError, response: RegisterStaticFilesResponse) => {
+                if (error) {
+                    resolve(errAsync(error));
+                } else {
+                    resolve(okAsync(response));
+                }
+            })
+        });
+        const resultRegisterStaticFiles: Result<RegisterStaticFilesResponse, Error> = await promiseRegisterStaticFiles;
+        if (!resultRegisterStaticFiles.isOk()) {
+            return err(resultRegisterStaticFiles.error);
+        }
+        const resp: RegisterStaticFilesResponse = resultRegisterStaticFiles.value;
+
+        const staticFileDestRelativeFilepathsMap: Map<string, string> = resp.getStaticFileDestRelativeFilepathsMap();
+        for (let [staticFileIdStr, destFilepathRelativeToEnclaveVolRoot] of staticFileDestRelativeFilepathsMap.entries()) {
+
+            const staticFileId: StaticFileID = <StaticFileID>(staticFileIdStr);
+
+            if (!staticFileFilepaths.has(staticFileId)) {
+                return err(new Error("No source filepath found for static file " + staticFileId + "; this is a bug in Kurtosis"));
+            }
+            const srcAbsFilepath: string = staticFileFilepaths[staticFileId];
+
+            const destAbsFilepath: string = path.join(this.enclaveDataVolMountpoint, destFilepathRelativeToEnclaveVolRoot);
+            
+            const promiseDestAbsFilepath: Promise<fs.Stats> = fsPromises.stat(destAbsFilepath);
```

Comment:
Same problem with `catch` here, and everywhere else you use it. Also, the naming here could prob be more explicit (`statDestAbsFilepathPromise`), though that's a minor thing